packages/client/mutations/SetTaskHighlightMutation.ts

```diff
@@ -0,0 +1,98 @@
+import {commitMutation} from 'react-relay'
+import graphql from 'babel-plugin-relay/macro'
+import {matchPath} from 'react-router-dom'
+import {Disposable, RecordSourceProxy} from 'relay-runtime'
+import {
+  SetTaskHighlightMutation as TSetTaskHighlightMutation,
+  SetTaskHighlightMutationVariables
+} from '../__generated__/SetTaskHighlightMutation.graphql'
+import {LocalHandlers, SharedUpdater} from '../types/relayMutations'
+import Atmosphere from '../Atmosphere'
+import {SetTaskHighlightMutation_meeting} from '~/__generated__/SetTaskHighlightMutation_meeting.graphql'
+import {ITask} from '../types/graphql'
+import {ClientNewMeeting} from '~/types/clientSchema'
+
+graphql`
+  fragment SetTaskHighlightMutation_meeting on SetTaskHighlightPayload{
+    meetingId
+    taskId
+    isHighlighted
+  }
+`
+
+const mutation = graphql`
+  mutation SetTaskHighlightMutation(
+    $taskId: ID!
+    $meetingId: ID!
+    $isHighlighted: Boolean!
+  ) {
+    setTaskHighlight(
+      taskId: $taskId
+      meetingId: $meetingId
+      isHighlighted: $isHighlighted
+    ) {
+      ...SetTaskHighlightMutation_meeting @relay(mask: false)
+    }
+  }
+`
+
+interface UpdaterOptions {
+  atmosphere: Atmosphere
+  store: RecordSourceProxy
+}
+
+// used only by subscription
+export const setTaskHighlightUpdater: SharedUpdater<SetTaskHighlightMutation_meeting> = (
+  payload,
+  {store}: UpdaterOptions
+) => {
+  const meetingId = payload.getValue('meetingId')
+  const {pathname} = window.location
+  const meetingRoute = matchPath(pathname, {path: `/meet/${meetingId}`})
+  /*
+   * Avoid adding task highlights for clients that are not in the meeting
+   */
+  if (!meetingId || !meetingRoute) {
+    return
+  }
+  const meeting = store.get<ClientNewMeeting>(meetingId)
+  if (!meeting) {
+    return
+  }
+
+  const highlightedTaskId = meeting.getValue('highlightedTaskId')
+  const taskId = payload.getValue('taskId')!
+  const isHighlighted = payload.getValue('isHighlighted')
+
+  // If we missed the un-highlighting of the previous task, then clear it here.
```

Comment:
I wasn't able to reproduce this. Are there repro steps I can take or does it just happen occasionally?

Reply:
This is just a precaution, it could happen when losing the connection for example.